
Titanic Data Storytelling: Exploring Survival Patterns
1. Dataset Introduction
Text Cell
The Titanic dataset contains information about passengers aboard the RMS Titanic. It includes details such as age, gender, passenger class, fare paid, and survival status. The objective is to analyze the data and identify factors that influenced passenger survival.

In [1]:
from google.colab import files
uploaded = files.upload()

Saving titanic.csv to titanic.csv


In [2]:
import pandas as pd

df = pd.read_csv('titanic.csv')
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [4]:
df.isnull().sum()

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,177
SibSp,0
Parch,0
Ticket,0
Fare,0


In [5]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


Observations:
The dataset contains 891 passengers.
Some missing values exist in Age, Cabin, and Embarked.
Passenger ages vary widely.
Fare values show significant variation.

In [6]:
df['Age'] = df['Age'].fillna(df['Age'].mean())
df['Embarked'] = df['Embarked'].fillna(df['Embarked'].mode()[0])

df.isnull().sum()

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,0
SibSp,0
Parch,0
Ticket,0
Fare,0


Missing values in Age and Embarked were handled using mean and mode imputation.

In [7]:
import plotly.express as px

fig = px.histogram(df, x='Survived',
                   title='Passenger Survival Count')
fig.show()

In [8]:
fig = px.pie(df,
             names='Sex',
             title='Gender Distribution')
fig.show()

In [9]:
fig = px.histogram(df,
                   x='Age',
                   title='Age Distribution')
fig.show()

In [10]:
fig = px.histogram(df,
                   x='Pclass',
                   title='Passenger Class Distribution')
fig.show()

In [11]:
fig = px.scatter(df,
                 x='Age',
                 y='Fare',
                 color='Survived',
                 title='Age vs Fare')
fig.show()

Key Findings:
Female passengers had a higher survival rate than male passengers.
First-class passengers were more likely to survive.
Most passengers were between 20 and 40 years old.
Higher fare-paying passengers generally had better survival chances.
Passenger class significantly influenced survival outcomes.

Conclusion:
The analysis shows that gender, passenger class, and fare played important roles in determining survival during the Titanic disaster. Female passengers and first-class travelers had a greater probability of survival. Data storytelling helps transform raw data into meaningful insights that can support decision-making and understanding of historical events.